### PR TITLE
Pass missing signal code for `child_process.spawnSync`

### PR DESCRIPTION
**Pull request by ippsav**
### What does this PR do?

Fixes(https://github.com/oven-sh/bun/issues/15111)

### CODE REVIEW

**Review by cirospaciari — CHANGES_REQUESTED** [review]
We need tests so we know the issue is fixed on CI in all platforms.

**Review by cirospaciari — APPROVED** [review]

**Comment by ippsav** [comment]
> > @cirospaciari I've added a simple test to replicate the issue, let me know if there are any more changes required
> 
> Thank you lets run the CI, running locally looks good.

I think i should wrap the code with an if to check if it's not Windows, since Windows doesn't support Posix signals

File: src/js/node/child_process.ts [hunk 1]
```diff
@@ -573,7 +574,7 @@ function spawnSync(file, args, options) {
   });
 
   const result = {
-    signal: null,
+    signal: signalCode,
```
Comment (cirospaciari) [inline src/js/node/child_process.ts]: Looking the tests and testing locally against node.js we should return null here if signalCode is undefined, you can it.if to not run on windows, we already have the test `test/js/node/test/parallel/unhandled-exception-with-worker-inuse.test.js` that will check for the null case

**Review by cirospaciari — CHANGES_REQUESTED** [review]

File: src/js/node/child_process.ts (hunk 1 above)
Comment (ippsav) [reply src/js/node/child_process.ts]: good catch it should be null now in the case of no `signalCode` returned, and thanks for the tip didn't know `it.if(condition)` is a thing that's awesome

Comment (cirospaciari) [reply src/js/node/child_process.ts]: Running the CI again lets see if the tests get something, looks good now.

**Review by cirospaciari — APPROVED** [review]

**Comment by cirospaciari** [comment]
Thank you

**Pull request merged**